gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -347,6 +361,94 @@ gum_process_enumerate_ranges (GumPageProtection prot,
 }
 
 void
+gum_process_enumerate_malloc_ranges (GumFoundMallocRangeFunc func,
+                                     gpointer user_data)
+{
+  task_t task;
+  kern_return_t ret;
+  unsigned i;
+  vm_address_t * malloc_zone_addresses;
+  unsigned malloc_zone_count;
+
+  task = mach_task_self ();
+
+  ret = malloc_get_all_zones (task,
+      gum_read_malloc_memory, &malloc_zone_addresses,
+      &malloc_zone_count);
+  if (ret != KERN_SUCCESS)
+    return;
+
+  for (i = 0; i != malloc_zone_count; i++)
+  {
+    vm_address_t zone_address;
+    malloc_zone_t * zone;
+
+    zone_address = malloc_zone_addresses[i];
+    zone = (malloc_zone_t *) zone_address;
+
+    if (zone && zone->introspect && zone->introspect->enumerator) {
```

Comment:
Brace should go to the next line. Also, when checking pointers (and not gboolean/bool) values, we always do `zone != NULL` instead of `zone`, so it would be good to also do that here.